enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/DocumentVerificationRepository.java

```diff
@@ -47,16 +48,10 @@ public interface DocumentVerificationRepository extends JpaRepository<DocumentVe
             "AND d.status IN (:statuses)")
     int failVerifications(String activationId, Date timestamp, List<DocumentStatus> statuses);
 
-    @Modifying
-    @Query("UPDATE DocumentVerificationEntity d " +
-            "SET d.status = com.wultra.app.enrollmentserver.model.enumeration.DocumentStatus.FAILED, " +
-            "    d.usedForVerification = false, " +
-            "    d.errorDetail = :errorMessage, " +
-            "    d.errorOrigin = :errorOrigin, " +
-            "    d.timestampLastUpdated = :timestamp " +
+    @Query("SELECT d.id FROM DocumentVerificationEntity d " +
             "WHERE d.timestampLastUpdated < :cleanupDate " +
             "AND d.status IN (:statuses)")
-    int failExpiredVerifications(Date cleanupDate, Date timestamp, String errorMessage, ErrorOrigin errorOrigin, List<DocumentStatus> statuses);
+    List<String> findExpiredVerifications(Date cleanupDate, List<DocumentStatus> statuses);
```

Comment:
What happens in case the query returns a lot of results? Will everything else work properly?

Reply:
Good catch, I will improve the scenario with many results.

Reply:
There is a hand break for Oracle because of limit at IN clause `for (List<String> idsChunk : Lists.partition(ids, BATCH_SIZE)) {` Theoretically, the transaction may expire, but the job runs quite often, so it should not handle thousands entries.
Let's postpone performance improvement, I will create a draft issue at least. Problem is, that JPQL doest not support LIMIT, native query is vendor specific and when using stream, we must process each entry separately, not in batch 😞

Reply:
Agreed, under normal conditions, this should not be an issue. It will only happens once the app breaks down and then everything will just keep getting worse because the cleanup jobs will also not work. Anyway, this should not happen.